src/drivers/snapdragon_pwm_out/snapdragon_pwm_out.cpp

```diff
@@ -0,0 +1,608 @@
+			/* sleep a bit before next try */
+			usleep(10000);
+			continue;
+		}
+
+
+		/* get controls for required topics */
+		unsigned poll_id = 0;
+
+
+		for (uint8_t i = 0; i < actuator_controls_s::NUM_ACTUATOR_CONTROL_GROUPS; i++) {
+
+			if (_controls_subs[i] >= 0) {
+
+				if (_poll_fds[poll_id].revents & POLLIN) {
+
+					orb_copy(_controls_topics[i], _controls_subs[i], &_controls[i]);
+
+				}
+
+				poll_id++;
+			}
+		}
+
+		if (_mixer != nullptr) {
+
+
+			_outputs.timestamp = hrt_absolute_time();
+
+			/* do  mixing for virtual control group */
+			_outputs.noutputs = _mixer->mix(_outputs.output,
+							_outputs.NUM_ACTUATOR_OUTPUTS,
+							NULL);
+
+
+			/* disable unused ports by setting their output to NaN */
+			for (size_t i = _outputs.noutputs; i < _outputs.NUM_ACTUATOR_OUTPUTS; i++){
+
+				_outputs.output[i] = NAN;
+
+			}
+
+			const uint16_t reverse_mask = 0;
+			uint16_t disarmed_pwm[4];
+			uint16_t min_pwm[4];
+			uint16_t max_pwm[4];
+
+			for (unsigned int i = 0; i < 4; i++) {
+				disarmed_pwm[i] = _pwm_disarmed;
+				min_pwm[i] = _pwm_min;
+				max_pwm[i] = _pwm_max;
+			}
+
+			uint16_t pwm[4];
+
+
+			// TODO FIXME: pre-armed seems broken -> copied and pasted from pwm_out_rc_in: needs to be tested
+			pwm_limit_calc(_armed.armed,
+				       false/*_armed.prearmed*/,
+				       _outputs.noutputs,
+				       reverse_mask,
+				       disarmed_pwm,
+				       min_pwm,
+				       max_pwm,
+				       _outputs.output,
+				       pwm,
+				       &_pwm_limit);
+
+
+			if (_armed.lockdown) {
+				send_outputs_pwm(disarmed_pwm);
+
+			} else {
+				send_outputs_pwm(pwm);
+			}
+
+			if (_outputs_pub != nullptr) {
+				orb_publish(ORB_ID(actuator_outputs), _outputs_pub, &_outputs);
+
+			} else {
+				_outputs_pub = orb_advertise(ORB_ID(actuator_outputs), &_outputs);
+			}
+
+
+		} else {
+			PX4_ERR("Could not mix output! Exiting...");
+			_task_should_exit = true;
+		}
+
+
+		bool updated;
+		orb_check(_armed_sub, &updated);
+
+		if (updated) {
+			orb_copy(ORB_ID(actuator_armed), _armed_sub, &_armed);
+		}
+	}
+
+	pwm_deinitialize();
+
+	for (uint8_t i = 0; i < actuator_controls_s::NUM_ACTUATOR_CONTROL_GROUPS; i++) {
+		if (_controls_subs[i] >= 0) {
+				orb_unsubscribe(_controls_subs[i]);
+		}
+	}
+
+	orb_unsubscribe(_armed_sub);
+
+	_is_running = false;
+
+}
+
+void task_main_trampoline(int argc, char *argv[])
+{
+	task_main(argc, argv);
+}
+
+void start()
+{
+	ASSERT(_task_handle == -1);
+
+	_task_should_exit = false;
+
+	/* start the task */
+	_task_handle = px4_task_spawn_cmd("pwm_out_main",
+					  SCHED_DEFAULT,
+					  SCHED_PRIORITY_MAX,
+					  1500,
+					  (px4_main_t)&task_main_trampoline,
+					  nullptr);
+
+	if (_task_handle < 0) {
+		warn("task start failed");
+		return;
+	}
+
+}
+
+void stop()
+{
+	_task_should_exit = true;
+
+	while (_is_running) {
+		usleep(200000);
+		PX4_INFO(".");
+	}
+
+	_task_handle = -1;
+}
+
+void usage()
+{
+
+	PX4_INFO("usage: pwm_out start [-d pwmdevice] -m mixerfile");
```

Comment:
It should be `[-m mixerfile]`, right? Because it will use the default if none is supplied.

Also, the command is called `snapdragon_pwm_out`, not just `pwm_out`.